### DISCUSSION TITLE
No routing issue ?

### DISCUSSION THREAD

**Discussion by ciePIC** [mathieucarbou/YaSolR#29] [General & Support]
Hello,
Thanks for your work !
I'm currently searching for a simple solar router and your project seems to match exactly with my needs.
I'm trying to setup a first solution and if it work's as expected, I will support you to have the pro version.

I have installed the last firmware on an ESP32 board, activated ZCD, o1_dimmer, mqtt power, o1_auto_dimmer.
I see the current power and I can manualy control the dimmer but there is no auto routing. When power go to negative value, nothing appear on output and dashboard stay at 0%.
Make I an error ? is it an issue ?

Can you help me please ?
Best regards,

PS: I have undertand that you live in france ? me too so if it's simpler, we can speek in french.

**Comment by mathieucarbou** [comment]
Bonjour!

Est ce que vous êtes passé à travers la manuel pour faire les étapes pour calibrer ou rentrer la valeur de votre résistance ?
Également voir dans la page  `/config` si le routage auto est activé.

Le projet est actuellement en fin de développement: j'ai quelques trucs à finir et je vais releaser une version 1 avec des version pro d'essais pour 3-4 jours afin que tout le monde puisse avoir un essai de la version pro.

Si vous avez besoin d'une version pro d'essai maintenant pour faciliter vos tests je peux vous l'envoyer.

Le routage auto fonctionne bien, même en mode remote ou mqtt, avec un amis on a nos routeurs en fonction, lui 2, et utilise jsy, mqtt, etc. Donc c'est sûrement une histoire de paramétrage ;-)

**Reply by mathieucarbou** [reply]
Notes: le projet étant en fin de développement, si vous voyez des bugs ou améliorations à faire, même au niveau documentation, n'hésitez pas à les partager :-)

**Reply by mathieucarbou** [reply]
Je viens à l'instant de tester sur la version OSS publiée et ça fonctionne bien :-)

**Comment by ciePIC** [comment]
Bonjour,

Merci pour votre retour rapide.
J'ai lu le manuel mais je suis passé à coté de se paramètre, le croyait qu'il était auto calculé. Effectivement une fois que l'on renseigne la valeur de résistance, cela fonctionne.
Merci !

Je n'ai pas besoin de version d'essais pro pour le moment, je reviendrais vers vous si besoin.

Je profite de notre échange pour 2 choses
- Je suis très intéressé par la version Ethernet car beaucoup plus stable que mon réseau Wifi.
Lorsque je flash le firmware dans une carte WT32-ETH01, le système semble fonctionner (affichage de la page web) mais redémarre régulièrement
`[11:53:39]E (31599) task_wdt: Task watchdog got triggered. The following tasks/users did not reset the watchdog in time: `
`[11:53:39]E (31599) task_wdt:  - IDLE1 (CPU 1) `
`[11:53:39]E (31599) task_wdt: Tasks currently running: `
`[11:53:39]E (31599) task_wdt: CPU 0: IDLE0 `
`[11:53:39]E (31599) task_wdt: CPU 1: y-core `
`[11:53:39]E (31599) task_wdt: Aborting. `
`[11:53:39]E (31599) task_wdt: Print CPU 1 backtrace `
`[11:53:39]  `
`[11:53:39]  `
`[11:53:39]  `
`[11:53:39]  `
`[11:53:39]Backtrace: 0x4000bfed:0x3ffbc570 0x40092430:0x3ffbc580 0x4011048d:0x3ffbc5a0 0x400fae47:0x3ffbc5e0 0x400921b2:0x3ffbc600  `
`[11:53:39] `
`[11:53:39] `
`[11:53:39]  `
`[11:53:39] `
`[11:53:39]ELF file SHA256: 5ba5f2744cc991f5 `
`[11:53:39] `
`[11:53:39]E (15830) esp_core_dump_flash: Core dump flash config is corrupted! CRC=0x7bd5c66f instead of 0x0 `
`[11:53:39]E (15838) esp_core_dump_elf: Elf write init failed! `
`[11:53:39]E (15843) esp_core_dump_common: Core dump write failed with error=-1 `
`[11:53:39]Rebooting...` 

- Avez-vous pour projet de permettre le paramétrage d'une IP fixe ? Ce serait vraiment un plus car mon réseau domotique n'a pas de DHCP. Cela m'éviterais donc une règle de pare-feu entre mes différents réseau.

Je suis actuellement très occupé niveau perso et pro mais a l'avenir, travaillant dans le domaine de l'électronique embarqué et de l'applicatif Web, il n'est pas exclu que je puisse vous aidez si vous avez besoin.

Cordialement,

**Reply by mathieucarbou** [reply]
Mercie!

Je vais regarder pour le twdt. Je n'ai pas testé la wt32 depuis que je l'ai ajouté et d'après votre trace il se peut que l'Ethernet demande beaucoup de cpu et impacte les autres tâches.

Pour l'ip fixe oui il faut que je l'ajoute c'est sans la liste des choses à faire.

**Comment by mathieucarbou** [comment]
@ciePIC : le support Ip fixe est ajouté: https://yasolr.carbou.me/manual#network-section

Par rapport à l'erreur, il s'agit du TWDT: c'est à dire que le watchdog redémarre l'ESP car il a détecté un ralentissement causé par une tache qui empeche les autres de s'exécuter.

Afin de déterminer vers ou est le problème j'aurais besoin de savoir ce que vous avez activé comme hardware: mqtt ? ds18 ? etc. Je suspecte que l'ajout d'une couche Eth ralentit énormément le processing réseau (mqtt et ou websocket).

Merci!

**Comment by ciePIC** [comment]
Merci pour le support IP fixe.
Je viens de faire la MAJ et cela fonctionne parfaitement.
Pour info, j'ai un réseau complexe avec un masque en /16 (255.255.0.0) et cela fonctionne sans soucis. Je vous donne cette info car j'ai rencontré des routeurs pro qui ne fonctionnait pas dans ce genre de config... bref...

Concernant la config, c'est celle de base. Les log que je vous ai transmit sont "out of the box". J'ai flashé le WT32-ETH01 puis je l'ai redémarré avec juste le câble réseau connecté et la console en UART (port de programmation). Je n'ai absolument pas touché a la config et je n'ai d'ailleurs même pas essayé de charger la webAPP. (test effectué sur un réseau de box classique en /24 donc sans complexité supplémentaire).

Je faire de faire un essais en déconnectant le câble réseau (uniquement alim + console UART) : 
`[19:54:51]ets Jun  8 2016 00:22:57`
`[19:54:51]`
`[19:54:51]rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)`
`[...]` <= Je n'ai pas mis tout les log pour racourcir le message. Si vous voulez plus de log, pas de soucis, je peut faire un record sur plusieurs minutes...
`[19:54:52][   456][I][ESPCONNECT][1][y-core] Starting Ethernet...`
`[19:55:06]E (29462) task_wdt: Task watchdog got triggered. Th,`
`[19:55:06]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00`
`[19:55:06]mode:DIO, clock div:1`
`[19:55:06]load:0x3fff0030,len:4832`
`[19:55:06]load:0x40078000,len:16460`
`[19:55:06]load:0x40080400,len:4`
`[19:55:06]load:0x40080404,len:3504`
`[19:55:06]entry 0x400805cc`
`[19:55:07]E (559) esp_core3յ}͡rɕ"յѥѥ2չjRT	SS%HY.\5KWE}͡rɕ"յѥѥ2չ[    14][W][YASOLR][1M[loopTask] Booting YaSolR OSS main_73d9e75`
`[19:55:07][    14][I][SYSTEM][1][loopTask] Initializing File System...`

Merci et bonne soirée,

**Reply by mathieucarbou** [reply]
Merci pour l'info! Ça m'aide énormément je pense que je sais d'ou vient le probleme!

**Reply by mathieucarbou** [reply]
L'ETH du WT32 est très lent (je préfère la Lilygo ETH LITE S3). Lorsque le cable est connecté, des interruptions se déclenches avec un échange DHCP qui bloque tout le CPU. Il faut que je démarre le watchdog plus tard. Je devrais pouvoir reproduire assez facilement. Merci!

**Reply by ciePIC** [reply]
Je cherche à utiliser une carte WT32-ETH01 car j'en avais une en stock sur étagère mais si le driver LAN8720 pose trop de soucis qui n'existe pas avec le W5500 de la T-ETH-Lite ESP32 S3, je peux commander une carte lilygo et changer de carte, cela ne me dérange pas.
De plus, si cela est plus stable pour les futures versions, ce n'est que mieux...
Je vous laisse me dire ce qui vous arrange.
Bonne soirée

**Reply by mathieucarbou** [reply]
> Je cherche à utiliser une carte WT32-ETH01 car j'en avais une en stock sur étagère mais si le driver LAN8720 pose trop de soucis qui n'existe pas avec le W5500 de la T-ETH-Lite ESP32 S3, je peux commander une carte lilygo et changer de carte, cela ne me dérange pas. De plus, si cela est plus stable pour les futures versions, ce n'est que mieux... Je vous laisse me dire ce qui vous arrange. Bonne soirée

Merci c'est gentil! 
Gardez là, c'est un use case intéressant et c'est vraiment un bug que vous avez trouvé... 
Je viens de le fixer, les nouveaux firmware sont en cours de construction... ;-)

**Reply by ciePIC** [reply]
Merci beacoup, je viens de faire l'essais de mon coté et cela fonctionne maintenant sur ma WT32-ETH01.
Je vais continuer mes essais et je reviendrais vers vous si je rencontre un souci ou lorsque je passerais à la version pro.

**Reply by mathieucarbou** [reply]
Je vais faire un changement prochainement comme j'arrive à la limite des 2Mo de flash pour les ESP qui ont 4Mo donc il faudra exceptionnellement re-flasher à partir du PC: la prochaine mise à jour ne passera pas par OTA. Je dois changer l'approche et mettre ne place une petit partition Safeboot via laquelle faire le flash OTA car l'appli devient de plus en plus grosse.
